### PR TITLE
YQ-2068 LRU cache for TUnboxedValue

### PR DESCRIPTION
**Pull request by zverevgeny**
### Changelog category
* Not for changelog (changelog entry is not required)

### CODE REVIEW

File: ydb/library/yql/minikql/mkql_key_payload_value_lru_cache.h [hunk 1]
```diff
@@ -0,0 +1,127 @@
+#pragma once
+
+#include <ydb/library/yql/minikql/computation/mkql_computation_node_pack.h>
```
Comment (rvu1024) [inline ydb/library/yql/minikql/mkql_key_payload_value_lru_cache.h:3]: лишний

**Review by rvu1024 — CHANGES_REQUESTED** [review]

File: ydb/library/yql/minikql/mkql_key_payload_value_lru_cache.h [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+#include <ydb/library/yql/public/udf/udf_value.h>
+#include <util/generic/string.h>
+#include <unordered_map>
+#include <list>
+#include <chrono>
+
+namespace NYql::NDq {
```
[the remark this answers is not in the document]
Comment (zverevgeny) [reply ydb/library/yql/minikql/mkql_key_payload_value_lru_cache.h]: fixed

File: ydb/library/yql/minikql/mkql_key_payload_value_lru_cache.h (hunk 1 above)
Comment (zverevgeny) [reply ydb/library/yql/minikql/mkql_key_payload_value_lru_cache.h]: fixed

File: ydb/library/yql/minikql/mkql_key_payload_value_lru_cache.h [hunk 3]
```diff
@@ -0,0 +1,128 @@
+#pragma once
```
Comment (vitstn) [inline ydb/library/yql/minikql/mkql_key_payload_value_lru_cache.h:1]: please move to a minikql/computation

File: ydb/library/yql/minikql/mkql_key_payload_value_lru_cache.h [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+#include <chrono>
+
+namespace NKikimr::NMiniKQL {
+
+//Design notes:
+//  Commodity LRUCache with HashMap and DoubleLinkedList. Key(that is TUnboxedValue) is stored in both HashMap and List
+//  Lazy(postponed) TTL implementation. An entry is not deleted immediately when its TTL is expired, the deletion is postponed until:
+//   - when it is accessed via Get()
+//   - when it is not accessed for a long period of time and becomes the LRU. Then it is garbage collected in Tick()
+//  In order to store values in HashMap serializes Key(TUnboxedValue) to TString
```
Comment (vitstn) [inline ydb/library/yql/minikql/mkql_key_payload_value_lru_cache.h:18]: seems it's wrong. NUdf::TUnboxedValue is a key for HashMap

File: ydb/library/yql/minikql/mkql_key_payload_value_lru_cache.h (hunk 3 above)
Comment (zverevgeny) [reply ydb/library/yql/minikql/mkql_key_payload_value_lru_cache.h]: done

File: ydb/library/yql/minikql/mkql_key_payload_value_lru_cache.h (hunk 4 above)
Comment (zverevgeny) [reply ydb/library/yql/minikql/mkql_key_payload_value_lru_cache.h]: obsolete comment, removed

**Review by vitstn — APPROVED** [review]

**Review by rvu1024 — APPROVED** [review]

**Pull request merged**